app/controllers/api/v1/application_controller.rb

```diff
@@ -61,6 +67,27 @@ def current_api_user
         @current_api_user ||= authenticate
       end
 
+      def planning_application
+        scope = current_local_authority.planning_applications
+        param = if params.key? :reference
+          params[:reference]
+        elsif params.key? :planning_application_id
+          params[:planning_application_id]
+        else
+          params[:id]
+        end
+
+        if param.match?(/[a-z]/i)
+          scope.where(reference: param).first
```

Comment:
We've previously been sort of inconsistent about how we look up the application, so I've mostly tried to retain those inconsistencies here rather than completely unifying them.

Mainly, that is, we've got a few controllers which expect to have `nil` returned and some which expect to have a `RecordNotFound` exception raised.

I've got a better handle on them now (now that all the tests pass in general) so I could have a second look over them to see if there are places where that could still be improved.

Also, as you say, we're wanting to get rid of the v1 API so I didn't want to spend too much time improving it.

Reply:
Looks like I'd mixed up two issues (some controllers expecting nil not `RecordNotFound` and some controllers using `where.first` instead of `find_by`). The second is easy to fix (but we can't use `find_by!` without a larger rewrite because it'll raise the exception).

Also was able to get rid of one of the multiple locations we do the lookup, which simplifies things somewhat.